### DISCUSSION TITLE
Migrating from 4.11.2 to 5.4.2, private backing fields, serialization

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#2279] [Questions]
I'm in the process of moving all of our code from 4.11.2 to 5.4.2. One of the big hurdles is the 5,500 properties that still use `RelationshipTypes.PrivateField`. Most of these are in two solutions which are not under active development, but are updated once a year for each client when a new version of AutoCAD is released. They are Windows Forms apps connecting directly to SQL Server / Azure.

It would not be trivial to convert all of these to managed backing fields and there is no budget to do so at this stage. Can we set the the `SerializationFormatter` to `"BinaryFormatter"` and kick the can down the road?

Are there any other breaking changes or things I should be aware of? Was there a blog post when 5.0 was released?

**Comment by rockfordlhotka** [comment]
You can use BinaryFormatter as long as you aren't using .NET 5+ or Blazor, where it is deprecated and/or just not there.

Another alternative is to implement some code in a custom `BusinessBase` class (between `Csla.BusinessBase` and your business classes) that uses reflection to implement the get/set state methods. There was a discussion here (or on Discord?) within the past few weeks about this, and I wrote some sample code to demonstrate the concept.

**Reply by rockfordlhotka** [reply]
Here's one issue that was filed out of that discussion: https://github.com/MarimerLLC/csla/issues/2148

**Reply by rockfordlhotka** [reply]
Somewhere along the way in v4.x and certainly in v5, I started relying on releasenotes.md to maintain lists of breaking changes and so forth.

Here's the ongoing [v5 releasenotes.md file](https://github.com/MarimerLLC/csla/blob/v5.x/releasenotes.md).

Look for the 🛑 and ⚠ icons for potential trouble spots.
